src/uu/hashsum/src/hashsum.rs

```diff
@@ -586,44 +587,37 @@ fn digest_reader<'a, T: Read>(
     // Digest file, do not hold too much in memory at any given moment
     let windows = cfg!(windows);
     let mut buffer = Vec::with_capacity(524_288);
-    let mut vec = Vec::with_capacity(524_288);
-    let mut looking_for_newline = false;
     loop {
         match reader.read_to_end(&mut buffer) {
             Ok(0) => {
                 break;
             }
             Ok(nread) => {
                 if windows && !binary {
-                    // Windows text mode returns '\n' when reading '\r\n'
-                    for &b in buffer.iter().take(nread) {
-                        if looking_for_newline {
-                            if b != b'\n' {
-                                vec.push(b'\r');
-                            }
-                            if b != b'\r' {
-                                vec.push(b);
-                                looking_for_newline = false;
-                            }
-                        } else if b != b'\r' {
-                            vec.push(b);
-                        } else {
-                            looking_for_newline = true;
-                        }
+                    // In Windows text mode, replace each occurrence of
+                    // "\r\n" with "\n".
+                    //
+                    // Find all occurrences of "\r\n", inputting the
+                    // slice just before the "\n" in the previous
+                    // instance of "\r\n" and the beginning of this
+                    // "\r\n".
+                    //
+                    // FIXME This fails if one call to `read()` ends
```

Comment:
Do you want to fix this in this PR, or in a follow up PR?

Reply:
In this pull request, I was intending to only remove the extra `vec`, so I would leave any other improvements to a different pull request. (That's a personal preference to solve one problem at a time with each pull request, but I'm happy to change the way I work to fit the project.)

Technically, the FIXME comment is incorrect: because of the use of `read_to_end()` instead of `read()`, this situation can never actually happen (since the entire file is read, and the `loop` has only one iteration). I put the comment there in case someone subsequently changes the `read_to_end()` to `read()`, as mentioned in #2295.

Reply:
That makes sense! I was just a bit hesitant to merge a PR that introduces a bug that did not exist before.
>  solve one problem at a time

That's a good way to work, thanks for your contributions!